### PR TITLE
[webui] Make feature switch based on current User 

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

**Comment by ChrisBr** [comment]
Splitted this PR into another one https://github.com/openSUSE/open-build-service/pull/3478
as this one probably needs some more discussions.

One problem with this approach is that we use the Feature.active? check during initilization of the app (routes and assets), this won't work anymore with this approach. However, we can easily check this dynamically in the controller.

File: src/api/lib/feature_switch/feature.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module Feature
+  def self.active?(feature)
+    if User.current && (User.current.is_admin? || User.current.is_staff?)
```
Comment (hennevogel) [inline src/api/lib/feature_switch/feature.rb]: How many database queries are those per request now?

File: src/api/lib/feature_switch/feature.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      data = Rails.cache.fetch('features', expires_in: 1.hour) do
+        YAML.load_file("#{Rails.root}/config/feature.yml").with_indifferent_access
```
Comment (hennevogel) [inline src/api/lib/feature_switch/feature.rb]: So I have to wait for one hour when I enable a feature?

File: src/api/lib/feature_switch/feature.rb (hunk 1 above)
Comment (ChrisBr) [inline src/api/lib/feature_switch/feature.rb]: Zero (more), as we already do this with every request for the Peek status bar in the webui_controller:
https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/webui_controller.rb#L251

File: src/api/lib/feature_switch/feature.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/lib/feature_switch/feature.rb]: Yes! However, currently we have to anyway restart the application to get features enabled as we have auto_refresh option disabled. I thought about implementing it the same way as the gem does but that would be a lot of duplicate code and not sure if this is worth the effort and just caching it for one hour.

> Subsequent calls to Feature will access the cached in-memory representation of the list of features. So changes to toggles in the underlying repository would not be reflected in the application until you restart the application or manually call

Comment (hennevogel) [inline src/api/lib/feature_switch/feature.rb]: But restarting the application will not restart the cache store.

File: src/api/lib/feature_switch/feature.rb (hunk 1 above)
Comment (hennevogel) [inline src/api/lib/feature_switch/feature.rb]: okay

File: src/api/lib/feature_switch/feature.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/lib/feature_switch/feature.rb]: Ok, implemented the same way as it is in the gem!

**Comment by ChrisBr** [comment]
@hennevogel moved now the Feature logic to the controller (except for the assets but I think it is fine to generate and include the assets nevertheless...)

File: src/api/spec/features/webui/image_templates_spec.rb [hunk 3]
```diff
@@ -36,21 +36,37 @@
...
+        login admin
+      end
 
-    scenario 'enabled' do
-      Feature.run_with_activated(:image_templates) do
-        visit root_path
-        expect(page).to have_link("New Image")
+      scenario 'disabled' do
+        Feature.run_with_deactivated(:image_templates) do
+          visit root_path
```
Comment (hennevogel) [inline src/api/spec/features/webui/image_templates_spec.rb]: This doesn't test the controller filters...

**Comment by ChrisBr** [comment]
[Make feature switch based on current User #3477](https://trello.com/c/ZR32dPWD/490-make-feature-switch-based-on-current-user-3477)

**Comment by bgeuken** [comment]
[Allow users to enable "BETA" options](https://trello.com/c/KgahLVpE/482-allow-users-to-enable-beta-options)

**Pull request merged**